cmd/utils.go

```diff
@@ -1,31 +1,73 @@
 package cmd
 
 import (
+	"fmt"
+	"log/slog"
+	"os"
 	"path"
+	"strings"
 	"time"
 
 	"github.com/google/uuid"
 	"github.com/internetarchive/Zeno/config"
 	"github.com/internetarchive/Zeno/internal/pkg/crawl"
 	"github.com/internetarchive/Zeno/internal/pkg/frontier"
+	"github.com/internetarchive/Zeno/internal/pkg/log"
 	"github.com/internetarchive/Zeno/internal/pkg/utils"
 	"github.com/paulbellamy/ratecounter"
-	"github.com/sirupsen/logrus"
 )
 
 // InitCrawlWithCMD takes a config.Flags struct and return a
 // *crawl.Crawl initialized with it
 func InitCrawlWithCMD(flags config.Flags) *crawl.Crawl {
 	var c = new(crawl.Crawl)
 
+	// Logger
+	var elasticSearchConfig *log.ElasticsearchConfig
+	elasticSearchURLs := strings.Split(flags.ElasticSearchURLs, ",")
+	if elasticSearchURLs[0] == "" {
+		elasticSearchConfig = nil
+	} else {
+		elasticSearchConfig = &log.ElasticsearchConfig{
+			Addresses:   elasticSearchURLs,
+			Username:    flags.ElasticSearchUsername,
+			Password:    flags.ElasticSearchPassword,
+			IndexPrefix: flags.ElasticSearchIndexPrefix,
+			Level:       slog.LevelDebug,
+		}
+	}
+
+	customLogger, err := log.New(log.Config{
+		FileOutput:               "zeno.log",
+		FileLevel:                slog.LevelDebug,
+		StdoutLevel:              slog.LevelInfo,
+		RotateLogFile:            true,
+		RotateElasticSearchIndex: true,
+		ElasticsearchConfig:      elasticSearchConfig,
+	})
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	c.Log = customLogger
+
+	go func() {
```

Comment:
Not a huge impact but it could be slightly better to handle the closing of this goroutine when Zeno is stopped. (using a waitgroup to make sure it properly finished before exiting Zeno etc)